### DISCUSSION TITLE
Using this with v-for

### DISCUSSION THREAD

**Discussion by juliandreas** [Akryum/vue-observe-visibility#198] [Q&A]
Is there a way to use this with a v-for loop?

Right now all elements in the loop use the same observer, but I kind of want the observer to hit individual elements in the v-for. Is it possible?

**Comment by pleek91** [comment]
I also just discovered this is the case. I need each element in the v-for to have their own observer so that when the user scrolls to that element the callback is run.

**Comment by madsh93** [comment]
Same here. This is especially true for mobiles, as some content may stretch deeper in the viewport.

**Comment by pleek91** [comment]
I was able to accomplish this by checking the `isVisible` argument in my callback. The `once` option made it seem like I could say "fire once for each instance of this component" but when its in a v-for it'll fire multiple times. So I just went with the `isVisible` and also made sure the code in my callback would only be run once by setting a flag variable to true after it was run the first time.

**Comment by zimzimzimzim** [comment]
@Akryum Do you intend to fix this?

**Comment by Akryum** [comment]
See #230
